Wiki and docs updates (#2101)
elastic/eui: FAQ.md

## What is the Elastic UI Framework?

> The Elastic UI Framework (EUI) is a design library in use at Elastic to build React applications that need to share our branding and look and feel. It distributes UI React components and static assets for use in building web layouts. Alongside the React components is a SASS/CSS layer that can be used independently on its own.

Reviewer: ```suggestion
The Elastic UI Framework (EUI) is a design library in use at Elastic to build React applications that need to share our branding, look and feel. It distributes UI React components and static assets for use in building web layouts. Alongside the React components is a SASS/CSS layer that can be used independently on its own.
```

Reviewer 2: This still looks weird to me. I think _look and feel_ is generally treated as a singular item which makes this mostly work, but it feels antithetical to our oxford comma usage.

Author: Changed all mentions of "look and feel" to "aesthetic" which is a word @cchaos and i have slowly been moving towards to describe this stuff. Helps remove the commas :)

## Can I use EUI?

> Yes, though be aware of the [license](LICENSE.md) as always. Although the roadmap and priorities are directed by our own usage within Elastic, we do attempt to make the platform generically useful for any React application and try to test for it.

Reviewer: ```suggestion
Yes, but be aware of the [license](LICENSE.md) as always. Although the roadmap and priorities are directed by our own usage within Elastic, we do attempt to make the platform generically useful for any React application and try to test for it.
```

## What is the versioning, releases and upgrade strategy?

> We use [semver](https://semver.org/) for versioning and use that to denote breaking changes in EUI upgrades. Traditionally we consider API changes in our prop names or component functionality to be a reason for a breaking change, but do not track the renaming of CSS selectors, mixins or other style changes under this same rigor.

Reviewer 2: ```suggestion
We use [semver](https://semver.org/) for versioning and use that to denote breaking changes in EUI upgrades. Traditionally we consider API changes in our prop names or existing component functionality to be a reason for a breaking change, but do not track the renaming of CSS selectors, mixins or other style changes under this same rigor.
```

> We use [semver](https://semver.org/) for versioning and use that to denote breaking changes in EUI upgrades. Traditionally we consider API changes in our propnames or component functionality to be a reason for a breaking change, but do not track the renaming of CSS selectors, mixins or other style changes under this same rigor.

Reviewer: ```suggestion
We use [semver](https://semver.org/) for versioning and use that to denote breaking changes in EUI upgrades. Traditionally we consider API changes in our prop names or component functionality to be a reason for a breaking change, but do not track the renaming of CSS selectors, mixins or other style changes under this same rigor.
```

> Traditionally releases are made weekly against whatever is in master and you can update from NPM as you see fit.

Reviewer 2: ```suggestion
Traditionally releases are made weekly against whatever is in master and you can upgrade from NPM as you see fit.
```

## How do you handle Typescript, Sass and theming?

> EUI started as a JS and Sass library that outputs a separate CSS and JS bundle. Over time we introduced TypeScript and started adding types, writing new components in TS, and migrating old components over. The goal is to actively convert everything over into a full TS system.

Reviewer 2: ```suggestion
EUI started as a JS and Sass library that outputs separate CSS and JS bundles. Over time we introduced TypeScript and started adding types, writing new components in TS, and migrating old components over. The goal is to actively convert everything over into a full TS system while retaining our build targets of ES5+commonjs, ES5+ES Modules, and a consolidated bundle file.
```

For styling we use Sass and generate a final CSS blob for the entire library, with some JSON theming files extracted from the Sass. You can find more information in [consuming EUI](wiki/consuming) and [theming](wiki/theming.md).

## Can I contribute to EUI

> Yes! We accept PRs regularly similar to our other Elastic repo.

Reviewer: ```suggestion
Yes! We accept PRs regularly similar to our other Elastic repos.
```

## Why is EUI open source?

Many of our products themselves are open source and rely upon this library to function. The Elastic UI Framework began as a folder of code in Kibana and we decided it could be used beyond that codebase. It exists as an independent library so that patterns can be shared across teams and design standards can be scaled across our organization. Since most of our products are open source, we treat this one similarly as far as public publishing and conversation even if its usage tends to focus more inward towards Elastic itself.

## What about reporting bugs and feature requests?

> Bug reports and feature requests are most welcome, but our roadmap is driven primarily by internal usage so understand that caveat.

Reviewer: ```suggestion
Bug reports and feature requests are most welcome, but our roadmap is driven primarily by internal usage.
```
